### ISSUE TITLE
Screensaver is not inhibited when using steam controller

### ISSUE THREAD

**Issue by seijikun** [ValveSoftware/Proton#287]
I've been playing the game Remember Me (228300) using Proton the last two days, and I have had the problem, that the screensaver is not inhibited as long as I am only using the steam controller.

Linux: openSUSE Tumbleweed - 4.18.0-1-default.
DE: KDE with two monitors (2560x1440)
CPU: i7-4770S
GPU: R9 290 using amdgpu with Mesa userspace

When using the steam-controller in steam's big picture mode, the displays stay on.
Using the steam-controller in native games also does seem to correctly inhibit the screensaver.

**Comment by andyklimczak** [comment]
which screensaver are you using, [xscreensaver](https://www.jwz.org/xscreensaver/)?

I've noticed this with xscreensaver, there might be something on that side you can do too

**Comment by seijikun** [comment]
Screensaver was probably not the exactly correct description. I meant KDE turning the screens off after 5 minutes of idle. ("Screen Energy Saving")
As far as I know, that and screensaver are inhibited by the same [window hint](https://people.freedesktop.org/~hadess/idle-inhibition-spec/)

I meant that idle is not inhibited, similar to [this](https://github.com/ValveSoftware/steam-for-linux/issues/2649), just proton-only.

**Comment by ghost** [comment]
I've disabled sleep mode during proton testing as a temporary workaround.

**Comment by howdev** [comment]
yes I do have the same problem. Proton does not stop screensaver/monitor DPMS. The screensaver is on but is in the background but screensaver should not run. And monitor still turns off while game running, have to move the mouse.

**Comment by Uradamus** [comment]
I'm getting this as well with a generic 360 controller. I'm on Solus Mate; on the plus side Mate at least has an applet for inhibiting power saver features (Inhibit Applet). I just added that to my panel for now and use it to toggle the power saver off before I start playing a game with a controller.

**Comment by mlassnig** [comment]
Also happens to me (i.e., KDE Plasma, xbox360 gamepad, DarkSouls3). Had to resort to this as a workaround: https://github.com/foresto/joystickwake#readme

**Comment by nstetter** [comment]
Can confirm.
KDE Plasma 5.13.5, Xbox360 Controller, Borderlands

**Comment by TooManyErrors** [comment]
Confirming this as well.
My computer attempted (and failed) to suspend after playing an hour of Monster Hunter World on my Steam Controller. To me, a non-programmer, it looks like having Proton utilize the D-Bus service org.freedesktop.ScreenSaver would solve the issue.

Running Ubuntu 18.10 with KDE Plasma 5.14.4.

**Comment by ziabice** [comment]
I'm having the same problem with a Logitech F310 gamepad and KDE 5.15.2 on Manjaro Linux, Proton 4.2-2. The issue happens also under wine-staging 4.5

**Comment by Bleuzen** [comment]
Best solution for now seems to be:
Use joystickwake.
Just install the `joystickwake-git` AUR package.

**Comment by Ahmed-E-86** [comment]
I use Caffeine extension with Gnome, and Feral GameMode with all desktops to keep screen awake during playing games (Native or Wine games).

To install GameMode:
`git clone https://github.com/FeralInteractive/gamemode.git`
`cd gamemode`
`./bootstrap.sh`

To use it with native games:
`gamemoderun ./game`

To use it with wine games:
`gamemoderun wine ./game`

To use it with Steam:

Add this to the Steam launch options:  `gamemoderun %command%`

**Comment by AwesamLinux** [comment]
I can confirm this is a problem as well with a Nintendo Switch Pro controller used over Bluetooth on Linux Mint

Distro: Linux Mint 19.1 Tessa (64 bit)
GPU: Vega 56
Driver/LLVM version: Mesa 19.1.0-devel/9.0.0
Kernel version: 5.1.0-050100-lowlatency
Proton version: 4.2-3

**Comment by zebulon2** [comment]
I confirm the issue with Shadow of War and the Steam Controller. Using gamemode from Feral however fixes the issue and nicely tunes performance.

**Comment by howdev** [comment]
Still not fixed for Proton 4.11-6? screensaver or dpms still comes on when the game is running.

**Comment by soredake** [comment]
Any progress on this?

**Comment by howdev** [comment]
the issue is long time ago. I don't have this issue anymore after Proton and steam updates

**Comment by rhysperry111** [comment]
I'm still experiencing this issue when playing NFS: Payback with my DS4 in Steam Input mode.

**Comment by DonKatsu** [comment]
I've been using gamemode since I started gaming on Linux, which explicitly inhibits monitor sleep.
Recently I haven't been using it and have since experienced Screen Energy Saving putting my monitors to sleep while playing with a controller. Happens native or with Proton.

While a game is running the Power Management applet shows "[Steam is currently blocking sleep and screen locking (Client Interface)](https://github.com/ValveSoftware/steam-for-linux/assets/8627704/de5a3a50-d8d9-42d7-ad56-2ef0be8c2b71)" but it doesn't seem to actually inhibit monitor sleep?
Running with gamemode again inhibits monitor sleep, with an entry of its own to the Power Management applet after Steam's.

On Fedora 38 using KDE with Wayland.

**Comment by walmartshopper** [comment]
I have the same issue on Arch / KDE / Wayland.  When I'm playing Rocket League with a Wii U Pro controller and Steam Input, the screen keeps turning off in the middle of games and it takes a few seconds for them to turn back on, which is enough to lose the game sometimes.

**Comment by DonKatsu** [comment]
Still an issue on Plasma 6.1.2 Wayland.
The Power Management applet still [shows Steam as blocking](https://github.com/user-attachments/assets/d18237aa-ca50-4c47-b9a8-89d0bb3b8429), yet the lock screen still triggers.
I have found that Lutris also fails to block locking/sleep with its "Disable screen saver" option despite the launched game showing in Power Management as blocking. Same implementation?
mpv and Youtube via Firefox successfully block locking/sleep. Toggling manual blocking works as well, of course.

**Comment by ahjolinna** [comment]
notice with when playing Clair Obscur: Expedition 33 (+dualsense controller) that this hasn't been fixed yet, I'm using:

```
Operating System: openSUSE Tumbleweed 20250505
KDE Plasma Version: 6.3.80
KDE Frameworks Version: 6.15.0
Qt Version: 6.9.0
Kernel Version: 6.12.25-1-longterm (64-bit)
Graphics Platform: Wayland
Processors: 16 × AMD Ryzen 7 5700G with Radeon Graphics
Memory: 48 GiB of RAM (46.8 GiB usable)
Graphics Processor: NVIDIA GeForce RTX 3060 Ti
Manufacturer: ASUS
```
